templates/site/js/main.js

```diff
@@ -267,105 +209,65 @@ function deleteNodeUser(username, nodeName){
 
                 })
 }
 
 //Page functionality and page details functions
-
-//Matches base url of the current node with the node registry and sets the email for the reset password modal and
-//the error message for a login error
-
+//---------------------------------------------
 function setNodeAdminEmail(){
-    const githubURL = "{{ node_registry_url }}";
-    const sessionURLFragment = "magpie/session";
+    var nodeEmailHref = `mailto:{{ current_node_admin_email }}`;
     let passwordResetEmail = document.getElementById("passwordResetEmail");
-
-    getSessionDetails().then(sessionJSON => {
-        var currentNodeURL = sessionJSON.url.replace(sessionURLFragment, "");
-        fetch(githubURL).then(resp => resp.json()).then(json => {
-            var node_keys = Object.keys(json);
-            node_keys.forEach((key, index) => {
-                if (json[key].links) {
-                    json[key].links.forEach((link) => {
-                        if (link.rel && link.rel === "service") {
-                            if (currentNodeURL.includes(link.href)) {
-                                passwordResetEmail.setAttribute("href", "mailto:" + json[key].contact);
-                                passwordResetEmail.innerText = json[key].contact;
-                            }
-                        }
-                    })
-                }
-            })
-        })
-    })
+    passwordResetEmail.setAttribute("href", nodeEmailHref);
+    passwordResetEmail.innerText = `{{ current_node_admin_email }}`;
 }
 
-function setNodeName(node_url, elementID){
-
-    getNodeRegistry().then(json => {
-        var node_keys = Object.keys(json);
-        let node_info;
+function setCaptionNodeName(element_id){
+    var captionElement = document.getElementById(element_id);
+    captionElement.innerText = `{{ current_node_name }}`;
+}
 
-        node_keys.forEach(key => {
-        node_info = json[key];
-        Object.entries(node_info).forEach(([node_details_key, node_details]) => {
+//Sets email link on an anchor tag
+function setNodeContact(elementID){
+    elementID.href = "mailto:" + `{{ current_node_admin_email }}`;
+    elementID.innerText = `{{ current_node_admin_email }}`;
+}
 
-                if(node_details_key == "links"){
-                    node_info[node_details_key].forEach((link) => {
+function getNodeServices(){
+    const servicesURLFragment = "/services";
 
-                        if(link.rel == "service"){
-                            if(node_url == link.href){
-                                elementID.innerText = node_info["name"];
-                            }
-                        }
-                    })
-                }
-            })
+    return fetch(servicesURLFragment,{
+        method: "GET",
+        headers: {
+            "Accept": "application/json, text/plain",
+            "Content-Type": "application/json"
+            }
+        }).then(response => {return response.json()}).then(json =>{
```

Comment:
The extra `.then(json => { return json })` is redundant.

`.then` returns a promise which is then returned by this function. When you use this function later on you access the result of the completed promise with `.then(json => { some code here })`. Since the promise at the end of this function does nothing other than returning its input, it is unnecessary.

Reply:
Removed the extra .then() portion